### ISSUE TITLE
[Question] Adldap\Auth\BindException: Can't contact LDAP server get Host

### ISSUE THREAD

**Issue by HepplerDotNet** [Adldap2/Adldap2-Laravel#674]
- Laravel Version: 5.7
- Adldap2-Laravel Version: 5.1.1
- PHP Version: 7.3
- LDAP Type: ActiveDirectory

Hi,
I'm using Adlap2-Laravel to provide some kind of self-service, which allows users to unlock their accounts.
We're using 12 Domain Controllers accross different locations, so I have written a Job to Unlock Users.
I'm looping over an array of DC Hosts and dispatch a Job for each host. Sometimes I see a Adldap\Auth\BindException: Can't contact LDAP server in the failed_jobs table. 
But the Exception doesn't specify which Host failed.

Any idea how to get the Host which failed?

**Comment by stevebauman** [comment]
Hi @HepplerDotNet, unfortunately not in the current release, but you will be able to in the next major release of Adldap2-Laravel (v6.0) which is coming in a couple weeks (on vacation at the moment until March 4th).

Logging is included with Adldap2 v10.0 which will be included with v6.0 of Adldap2-Laravel. You will also be able to register listeners on LDAP binding events to perform your own logging or logic depending on what occurs:

https://adldap2.github.io/Adldap2/#/events

I'll leave this open and close once this v6.0 has been released.

**Comment by stevebauman** [comment]
v6 has been released. You can now register a listener to retrieve the host / connection of the binding failure:

```php
$dispatcher = \Adldap\Adldap::getEventDispatcher();

$dispatcher->listen(Binding::class, function (Failed $event) {
    $event->connection->getName(); // Returns 'default'
    $event->connection->getHost(); // Returns 'ldap://192.168.1.1:389'
});
```
